pandas/io/parsers.pyi

```diff
@@ -59,7 +59,7 @@ def read_csv(
     low_memory: bool = ...,
     memory_map: bool = ...,
     float_precision: Optional[str] = ...,
-) -> TextParser: ...
+) -> DataFrame: ...
```

Comment:
This function can return either `TextParser` or `DataFrame`; what was broken here that needed this fix?

Reply:
TextParser is a function, not a class.
```python
def TextParser(*args, **kwds): ...
```
It can't be used as a type.

Reply:
Huh, the docs must be wrong. I just read the code, and I think it actually returns `TextFileReader`. https://github.com/pandas-dev/pandas/blob/5d65e0a3092f3c30169d0c0a3d0227e985104a56/pandas/io/parsers/readers.py#L423

Reply:
I looked at the code and tried to figure out what it was returning based on inspection. That's where I got `DataFrame` from. But I might be wrong.